### PR TITLE
use a vector as a backing array for DataF

### LINKED ISSUE #877: Missing support for similar / zeros for point spaces
MWE:

```julia
import ClimaCore: Domains,Geometry,Meshes,Spaces,Fields,Utilities

FT = Float64
column = Domains.IntervalDomain(
    Geometry.ZPoint{FT}(-10.0),
    Geometry.ZPoint{FT}(0.0);
    boundary_tags = (:top, :bottom)
);
mesh = Meshes.IntervalMesh(column; nelems = 10);
center_space = Spaces.CenterFiniteDifferenceSpace(mesh);
face_space = Spaces.FaceFiniteDifferenceSpace(center_space);
surface_space = Fields.level(
    face_space,
    Utilities.PlusHalf(Spaces.nlevels(face_space)-1)
);

Fields.zeros(surface_space) # breaks
Fields.Field(FT, surface_space) # breaks
```

Ported from #869.

### PR DESCRIPTION
**Pull request by simonbyrne**
Alternative to #882, fixes #877.

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Review by charleskawczynski — APPROVED** [review]

File: src/Spaces/pointspace.jl [hunk 1]
```diff
@@ -5,15 +5,15 @@ local_geometry_data(space::AbstractPointSpace) = space.local_geometry
...
+A zero-dimensional space.
 """
 struct PointSpace{LG} <: AbstractPointSpace
     local_geometry::LG
 end
 
 function PointSpace(local_geometry::LG) where {LG <: Geometry.LocalGeometry}
     FT = Geometry.undertype(LG)
-    local_geometry_data = DataLayouts.DataF{LG}(Matrix{FT})
+    local_geometry_data = DataLayouts.DataF{LG}(Array{FT})
```
Comment (charleskawczynski) [inline src/Spaces/pointspace.jl:16]: Shouldn't we be specifying the size? What will this do exactly?

Comment (charleskawczynski) [reply src/Spaces/pointspace.jl:16]: Ever since fixing inference issues in Thermodynamics, I've tried avoiding UnionAll types

Comment (simonbyrne) [reply src/Spaces/pointspace.jl:16]: just determines what type (Array, CuArray) and floating point type (Floa32, Float64). Dimensionality is handled by the constructor

**Comment by kmdeck** [comment]
this looks like the better solution! Though it is failing because it is a bit slower than previous code was?

**Comment by simonbyrne** [comment]
@charleskawczynski the failure looks unrelated: can we make the benchmark tests a little less strict?

**Comment by charleskawczynski** [comment]
> @charleskawczynski the failure looks unrelated: can you look into this, or make the benchmark tests a little less strict?

Yeah, we can make the benchmark less strict, I'll push a fix

**Pull request merged**